x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -254,7 +254,7 @@ public TokenService(Settings settings, Clock clock, Client client, XPackLicenseS
      * {@link #VERSION_TOKENS_INDEX_INTRODUCED} and to a specific security tokens index for later versions.
      */
     public void createOAuth2Tokens(Authentication authentication, Authentication originatingClientAuth, Map<String, Object> metadata,
-                                   boolean includeRefreshToken, ActionListener<Tuple<String, String>> listener) {
+                                   boolean includeRefreshToken, ActionListener<Tuple<Tuple<String, String>, Authentication>> listener) {
```

Comment:
Having Tuples of Tuples usually is a good sign that we need a new object for this

Reply:
How about Triple?

Reply:
I'd go with something like a 
```
public final class CreateTokenResult {
    private final String accessToken;
    private final String refreshToken;
    private final Authentication authentication;

    // ctor, getters etc...
}
```

Reply:
Updated the PR